service/src/main/java/de/adorsys/psd2/sandbox/xs2a/testdata/domain/Transaction.java

```diff
@@ -1,24 +1,31 @@
 package de.adorsys.psd2.sandbox.xs2a.testdata.domain;
 
-import java.math.BigDecimal;
 import java.time.LocalDate;
-import java.util.Currency;
 import lombok.AllArgsConstructor;
+import lombok.Data;
 import lombok.EqualsAndHashCode;
-import lombok.Getter;
 
 @AllArgsConstructor
-@Getter
+@Data
 @EqualsAndHashCode
 public class Transaction {
 
   private String transactionId;
-  private BigDecimal amount;
-  private Currency currency;
+  private String entryReference;
+  private Amount amount;
   private LocalDate bookingDate;
+  private LocalDate valueDate;
   private String debtorName;
-  private String debtorAccount;
+  private String debtorIban;
```

Comment:
Constructors with `String, String, String, String, String, ...` are error prone and hard to use. Maybe we can group some fields in an object or use a builder.

Reply:
will be done by some refactoring in the next weeks